server/protocols/webadmin/webadmin-mail-over-web/src/test/java/org/apache/james/request/MailPropsTest.java

```diff
@@ -0,0 +1,131 @@
+package org.apache.james.request;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.server.core.MailImpl;
+import org.apache.james.server.core.MimeMessageCopyOnWriteProxy;
+import org.apache.james.webadmin.request.BodyPartProps;
+import org.apache.james.webadmin.request.MailProps;
+import org.apache.mailet.Matcher;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class MailPropsTest {
+
+    @Test
+    public void mailImplemantationHascorrectFieldsWhenBuiltFromMailProps() throws Exception{
+        MailProps mailProps = new MailProps(
+                "test",
+                "valid@address",
+                Arrays.asList("valid1@address"),
+                Arrays.asList("valid2@address"),
+                Arrays.asList("valid3@address"),
+                "test-subject",
+                null,
+                null
```

Comment:
I cannot understand this test as there is way too many arguments to this constructor, and most of them have the same type.

We could have a builder?

Or the tested input can be a JSON (the test would then deserialize the JSON first then turn it into a mail?